dask/dataframe/groupby.py

```diff
@@ -1139,6 +1142,33 @@ def _build_agg_args_list(result_column, func, input_column):
     )
 
 
+def _build_agg_args_collect(result_column, _, input_column):
+    # NOTE: This function provides a cudf-compatible
+    # alternative to `_build_agg_args_list`
+    intermediate = _make_agg_id("collect", input_column)
+
+    return dict(
+        chunk_funcs=[
+            (
+                intermediate,
+                _apply_func_to_column,
+                dict(column=input_column, func=lambda s: s.agg("collect")),
```

Comment:
lambdas have a significant overhead when serialising, so we should avoid them whenever possible

Reply:
Okay cool. I agree with this, but was trying to mimic the list logic. I will revise.